src/client.rs

```diff
@@ -1117,18 +1183,36 @@ impl AudioHandler {
         let timeout = None;
         let stream = device.build_output_stream(
             config,
-            move |data: &mut [T], _: &_| {
+            move |data: &mut [T], info: &cpal::OutputCallbackInfo| {
                 if !*ready.lock().unwrap() {
                     *ready.lock().unwrap() = true;
                 }
-                let mut lock = audio_buffer.lock().unwrap();
+
                 let mut n = data.len();
-                if lock.occupied_len() < n {
-                    n = lock.occupied_len();
+                let mut lock = audio_buffer.lock().unwrap();
+                let mut having = lock.occupied_len();
+                if having < n {
+                    let tms = info.timestamp();
+                    let how_long = tms.playback.duration_since(&tms.callback).unwrap();
```

Comment:
remove possible unwrap

Reply:
Don't understand.
The unwrap is impossible to lead panic, and the container function does not allow use ? operator.
Uses unwrap seems to be the most clear style

Reply:
Change to unwrap_or(0)